kuma/wiki/tasks.py

```diff
@@ -266,3 +268,57 @@ def build_sitemaps():
     index_path = os.path.join(settings.MEDIA_ROOT, 'sitemap.xml')
     with open(index_path, 'w') as index_file:
         index_file.write(parseString(sitemap_index).toxml())
+
+
+@task
+def index_documents(ids, index_pk, reraise=False):
+    """
+    Index a list of documents into the provided index.
+
+    :arg ids: Iterable of `Document` pks to index.
+    :arg index_pk: The `Index` pk of the index to index into.
+    :arg reraise: False if you want errors to be swallowed and True
+        if you want errors to be thrown.
+
+    .. Note::
+
+       This indexes all the documents in the chunk in one single bulk
+       indexing call. Keep that in mind when you break your indexing
+       task into chunks.
+
+    """
+    from kuma.wiki.models import Document
+
+    cls = WikiDocumentType
+    es = cls.get_connection('indexing')
+    index = Index.objects.get(pk=index_pk)
+
+    objects = Document.objects.filter(id__in=ids)
+    documents = []
+    for obj in objects:
+        try:
+            documents.append(cls.from_django(obj))
+        except Exception:
+            log.exception('Unable to extract/index document (id: %d)', obj.id)
+            if reraise:
+                raise
+
+    cls.bulk_index(documents, id_field='id', es=es, index=index.prefixed_name)
+
+
+@task
+def unindex_documents(ids, index_pk):
+    """
+    Delete a list of documents from the provided index.
+
+    :arg ids: Iterable of `Document` pks to remove.
+    :arg index_pk: The `Index` pk of the index to remove items from.
+
+    """
+    cls = WikiDocumentType
+    es = cls.get_connection('indexing')
+    index = Index.objects.get(pk=index_pk)
+
+    for pk in ids:
+        es.delete(index=index.prefixed_name, doc_type=cls.get_doc_type(),
```

Comment:
I think there is a way to bulk delete documents, so maybe add a `WikiDocumentType.bulk_delete` method and call that?